Mathlib/Util/LongNames.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Std.Data.HashMap
+import Mathlib.Lean.Expr.Basic
+
+/-!
+# Commands `#long_names` and `#long_instances`
+
+For finding declarations with excessively long names.
+-/
+
+open Lean Meta Elab
+
+private def isBlackListed (declName : Name) : CoreM Bool := do
+  if declName.toString.startsWith "Lean" then return true
+  let env ← getEnv
+  pure $ declName.isInternal'
+   || isAuxRecursor env declName
+   || isNoConfusion env declName
+  <||> isRec declName <||> isMatcher declName
+
+/--
+Retrieve all names in the environment satisfying a predicate.
+-/
+def allNames (p : Name → Bool) : CoreM (Array Name) := do
+  let mut names := #[]
+  for (n, _) in (← getEnv).constants.map₁.toList ++ (← getEnv).constants.map₂.toList do
+    if p n && !(← isBlackListed n) then
+      names := names.push n
+  pure names
+
+def Lean.Name.getModule (n : Name) : CoreM (Option Name) := do
```

Comment:
docstring?

It would also be nice if this and the previous definition was in some basic Lean utility file

Reply:
Moved. And in fact refactored.